File: event_handlers_vb/Form1.vb
Public Class Form1
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Me.TextBox1.Text = ""
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        Me.Label1.Text = "LAST MSG SENT: " + Me.TextBox1.Text
    End Sub

    Private Sub TextBox1_MouseEnter(sender As Object, e As EventArgs) Handles TextBox1.MouseEnter
        Me.TextBox1.BackColor = Color.Green
    End Sub

    Private Sub TextBox1_MouseLeave(sender As Object, e As EventArgs) Handles TextBox1.MouseLeave
        Me.TextBox1.BackColor = Color.Red
    End Sub

    Private Sub Label1_MouseHover(sender As Object, e As EventArgs) Handles Label1.MouseHover
        Me.Label1.Text = "LAST MSG SENT: "
    End Sub
End Class

File: event_handlers_cs/Form1.cs
namespace event_handlers_cs
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            this.textBox1.Text = "";
        }

        private void button2_Click(object sender, EventArgs e)
        {
            this.label1.Text = "LAST MSG SENT: " + this.textBox1.Text;
        }

        private void textBox1_MouseEnter(object sender, EventArgs e)
        {
            this.textBox1.BackColor = System.Drawing.Color.Green;
        }

        private void textBox1_MouseLeave(object sender, EventArgs e)
        {
            this.textBox1.BackColor = System.Drawing.Color.Red;
        }

        private void label1_MouseHover(object sender, EventArgs e)
        {
            this.label1.Text = "LAST MSG SENT: ";
        }

        private void label1_Click(object sender, EventArgs e)
        {

        }

        private void Form1_Load(object sender, EventArgs e)
        {

        }
    }
}